pennylane/optimize/adaptive.py

```diff
@@ -195,7 +195,10 @@ def step_and_cost(self, circuit, operator_pool, drain_pool=False, params_zero=Tr
                 if (gate.name == operation.name and gate.wires == operation.wires)
             ]
             for gate in repeated_gates:
-                operator_pool.remove(gate)
+                for i, op in enumerate(operator_pool):
+                    if op is gate:
+                        operator_pool.pop(i)
+                        break
```

Comment:
I think you can combine this with the previous list comprehension using something like
```python
operator_pool = [
    gate for gate in operator_pool 
    if all(
        gate.name != operation.name or gate.wires != operation.wires 
        for operation in circuit.tape.operations
    )
]
```